flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/FunctionCatalogOperatorTable.java

```diff
@@ -84,12 +89,12 @@ public void lookupOperatorOverloads(
 			return;
 		}
 
-		String name = opName.getSimple();
-		Optional<FunctionLookup.Result> candidateFunction = functionCatalog.lookupFunction(
-			FunctionIdentifier.of(name));
+		FunctionIdentifier identifier = toFunctionIdentifier(opName.names.toArray(new String[0]), catalogManager);
```

Comment:
How about we use the UnresolvedIdentifier here? I think we could adjust the `FunctionCatalog#lookupFunction` slightly and make it accept the `UnresolvedIdentifier`. Than this qualifying could happen within `FunctionCatalog`.

Reply:
I think there are some concerns from @bowenli86 .
I guess if we use `UnresolvedIdentifier`, we can not distinguish system functions and catalog functions.
CC: @bowenli86

Reply:
We can. It's just that the resolution would happen in the `lookupFunction`. Not externally. The same logic that now happens in `toFunctionIdentifier` would happen as the first step in `lookupFunction`.

Reply:
I agree with @dawidwys . What `toFunctionIdentifier` does is just qualify 2-parts to 3-parts and this can happen in `FunctionCatalog`, because it holds the `CatalogManager`.